### PR TITLE
china and russia write loop

### PR DESCRIPTION
**Pull request by gerbsec**
tested

### CODE REVIEW

File: tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go [hunk 1]
```diff
@@ -1,20 +1,29 @@
 /*
 ID: d769fb8c-2cf9-4c10-b5ef-7035468b6ffd
-NAME: Enforce an outbound firewall
-CREATED: 2023-01-24
+RULE: Enforce an outbound firewall
```
Comment (privateducky) [inline tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go:3]: should be name, not rule

File: tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
 */
 package main
 
 import (
-	"github.com/preludeorg/test/endpoint"
+	Endpoint "github.com/preludeorg/test/endpoint"
 )
 
+var ips = []string{
+	"77.88.55.80:443",
```
Comment (privateducky) [inline tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go:13]: can you add a comment to reference the fqdn that goes to each of these?

File: tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
+	for _, ip := range ips {
+		code := Endpoint.DialTCP(ip, "hello world")
+		if code == 0 {
+			println("[+] Message sent successfully to", ip)
+			continue
+		}
+		println("[-] Failed to send message to", ip)
 	}
-	Endpoint.Stop(100)
+	Endpoint.Stop(101)
```
Comment (privateducky) [inline tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go:26]: if both IP addresses are blocked, we want to exit with the appropriate code. 
let's go with 106, if we're capable of handling the condition of "connection refused/blocked". I want to make sure we can distinguish from a connection that fails due to "doesn't exist" vs a firewall blocked it.

**Review by privateducky — CHANGES_REQUESTED** [review]
stuff

File: tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ func test() {
 		}
 		println("[-] Failed to send message to", ip)
 	}
-	Endpoint.Stop(101)
+	Endpoint.Stop(106)
```
Comment (privateducky) [inline tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go:26]: ``Endpoint.DialTCP`` returns a 1 on all error cases.
can you distinguish in there between our dual cases: refused connection (due to firewall) and "other"?

File: tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go (hunk 3 above)
Comment (gerbsec) [reply tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go]: resolved, but in a different section of the code

File: tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go (hunk 4 above)
Comment (gerbsec) [reply tests/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd/d769fb8c-2cf9-4c10-b5ef-7035468b6ffd.go]: new updated DialTCP is now "NetworkTest" and we talked about this process on call

**Comment by bfuzzy1** [comment]
Initial approval on the code. Needs some more testing through probes in the am.

**Review by privateducky — APPROVED** [review]

**Pull request merged**